### PR TITLE
Call extraction script from musedev script

### PR DESCRIPTION
**Pull request by ftc**
Things to do:
- [x] compress graph files
- [x] call web service
- [ ] extract organization and app name from residue?
- [x] process response
Notes so far:
- re: https://github.com/cuplv/biggroum/issues/56  I moved these dockerfiles and scripts to `biggroum/musedev_deploy`
- the dockerfile in `biggroum/musedev_deploy` is minimal and mostly just runs `setup_docker.sh` to get things ready for the api
- I use Popen to call the graph extractor script to supress stdout (I think there may be a better way to do this)

### CODE REVIEW

**Comment by smover** [comment]
@ftc the code that calls the graph extractor should not have prints on the stdout.
I would avoid to call popen to call another python interpreter that calls popen again.

The setupdocker.sh script should not be part of the docker container but should be called inside the biggroumcheck.sh script. That is, we do not change the docker container that musedev give us but we install what we want when our script is invoked.

**Comment by ftc** [comment]
Ok sounds good, I can make both changes.

File: python/fixrgraph/extraction/run_extractor.py [hunk 1]
```diff
@@ -329,19 +329,28 @@ def __init__(self, in_dir,
...
-        logging.info("Executing %s" % " ".join(args))
-
-        # not pipe stdout - processes will hang
-        # Known limitation of Popen
-        proc = subprocess.Popen(args, cwd=cwd)
-        proc.wait()
+        write_log(log,repo,"Executing %s" % " ".join(args),"info")
+
+        if log is None:
+            #TODO: check if this is still the case
```
Comment (ftc) [inline python/fixrgraph/extraction/run_extractor.py]: @smover I piped the stdout and stderr into the logger so we can use them for the i/o discussed in the musedev meeting.  There is a note that piping stdout causes the process to hang.  I did not observe a hang when testing it.  Do you know what this comment may be about?  Can I remove it?

**Comment by ftc** [comment]
I refactored the run_extractor to use the log parameter that it takes and piped the output of calling the graph extractor to the log.  This way I don't need to use the popen hack from earlier to suppress stdout.  I am out of time for tonight but I will move the docker setup script and call it from the `run_script.sh` tomorrow.

File: python/fixrgraph/musedev/api.py [hunk 2]
```diff
@@ -123,16 +134,54 @@ def finalize(cmd_input):
...
+                javafiles.append(filePath)
+
+    # extract the graphs
+    extractor_jar = os.getenv("GRAPHEXTRACTOR")
+    try:
+        graphdir = tempfile.mkdtemp(".groum_test_extract_single")
+
+        # To call directly, uncomment the following lines
+        #TODO: get github org and repo name
+        extract_single.extract_single_class_dir(repo = ["unkown","unknown",cmd_input.commit],
```
Comment (ftc) [inline python/fixrgraph/musedev/api.py]: @smover, For now, I use the organization and app name as "unknown" and "unknown" which seems to work. I would like to get this from somewhere, perhaps the muse custom tool API.

**Comment by smover** [comment]
@ftc @phillip-wells the call to the service was not really done and tested 🤔
I'm fixing that.

**Pull request merged**